toolchain/parser/parser_handle_period.cpp

```diff
@@ -6,21 +6,24 @@
 
 namespace Carbon {
 
-// Handles PeriodAs variants.
+// Handles PeriodAs variants and ArrowExpression.
 // TODO: This currently only supports identifiers on the rhs, but will in the
 // future need to handle things like `object.(Interface.member)` for qualifiers.
-auto ParserHandlePeriod(ParserContext& context, ParseNodeKind node_kind)
-    -> void {
+auto ParserHandlePeriodOrArrow(ParserContext& context, ParseNodeKind node_kind,
```

Comment:
```suggestion
static auto ParserHandlePeriodOrArrow(ParserContext& context, ParseNodeKind node_kind,
```

Sorry, pre-existing issue, but probably good to fix.